pkg/controller/drivelvgcontroller/drivelvgcontroller.go

```diff
@@ -0,0 +1,365 @@
+	log := d.log.WithFields(logrus.Fields{
+		"method": "reconcileLVG",
+	})
+	var (
+		status = lvg.Spec.GetStatus()
+		health = lvg.Spec.GetHealth()
+		name   = lvg.GetName()
+	)
+	// If LVG status is failed or Health is not good, try to reset its AC size to 0
+	if status == apiV1.Failed || health != apiV1.HealthGood {
+		return ctrl.Result{}, d.resetACSizeOfLVG(name)
+	}
+	// If LVG is already presented on a machine but doesn't have AC, try to create its AC using annotation with
+	// VG free space
+	size, err := getFreeSpaceFromLVGAnnotation(lvg.Annotations)
+	if err != nil {
+		log.Errorf("Failed to get free space from LVG %v, err: %v", lvg, err)
+		if err == errTypes.ErrorNotFound {
+			err = nil
+		}
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{}, d.createLVGACIfNotExistsOrUpdate(lvg, size)
+}
+
+// reconcileDrive preforms logic for drive reconciliation
+func (d *Controller) reconcileDrive(ctx context.Context, drive *drivecrd.Drive) (ctrl.Result, error) {
+	var (
+		health  = drive.Spec.GetHealth()
+		status  = drive.Spec.GetStatus()
+		isClean = drive.Spec.GetIsClean()
+	)
+	switch {
+	case isClean && (health == apiV1.HealthGood && status == apiV1.DriveStatusOnline):
+		return d.createACIfNotExistOrUpdate(ctx, drive.Spec)
+	case health != apiV1.HealthGood || status != apiV1.DriveStatusOnline:
+		return d.handleDriveIsNotGood(ctx, drive.Spec)
+	case !isClean:
+		return d.createACIfNotExistOrUpdate(ctx, drive.Spec)
+	}
+	return ctrl.Result{RequeueAfter: time.Minute}, nil
+}
+
+// createACIfNotExistOrUpdate tries to create AC for drive or update its size if AC already exists
+func (d *Controller) createACIfNotExistOrUpdate(ctx context.Context, drive api.Drive) (ctrl.Result, error) {
+	log := d.log.WithFields(logrus.Fields{
+		"method": "createACIfNotExistOrUpdate",
+	})
+	driveUUID := drive.GetUUID()
+	size := drive.GetSize()
+	// if drive is not clean, size is 0
+	if !drive.GetIsClean() {
+		size = 0
+	}
+	ac, err := d.cachedCrHelper.GetACByLocation(driveUUID)
+	switch {
+	case err == nil:
+		lvg, err := d.cachedCrHelper.GetLVGByDrive(ctx, driveUUID)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		// If LVG exists for Drive, we delete drive AC, because CSI uses LVG AC
+		if lvg != nil {
+			if err := d.client.DeleteCR(context.WithValue(ctx, base.RequestUUID, ac.Name), ac); err != nil {
+				log.Errorf("Error during update AvailableCapacity request to k8s: %v, error: %v", ac, err)
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{}, err
+		}
+		// If ac is exists, update its size to drive size
+		if ac.Spec.Size != size {
+			ac.Spec.Size = size
+			if err := d.client.Update(context.WithValue(ctx, base.RequestUUID, ac.Name), ac); err != nil {
+				log.Errorf("Error during update AvailableCapacity request to k8s: %v, error: %v", ac, err)
+				return ctrl.Result{}, err
+			}
+		}
+		return ctrl.Result{RequeueAfter: RequeueDriveTime}, nil
+	case err == errTypes.ErrorNotFound:
```

Comment:
Is that possible? Before we create LVG we modify AC type...

Reply:
`ErrorNotFound` is related to drive AC, not to LVG AC